File: hard/prefix.lua
function pol(o, a, b)
  return (o == '*') and a * b or (o == '/') and a / b or a + b
end

for line in io.lines(arg[1]) do
  local op, num = {}, {}
  for i in line:gmatch("[*/+]") do op[#op + 1] = i end
  for i in line:gmatch("%d+") do num[#num + 1] = i end
  local res = num[1]
  for i = #op, 1, -1 do
    res = pol(op[i], res, num[#num - i + 1])
  end
  print(math.floor(res + 0.001))
end

File: hard/prefix.rb
def pol(o, a, b)
  case o
  when '*'
    return a * b
  when '/'
    return a / b
  end
  a + b
end

File.open(ARGV[0]).each_line do |line|
  s = line.chomp.split
  n = s.length / 2
  res = s[n].to_f
  (1..n).each do |x|
    res = pol(s[n - x], res, s[n + x].to_f)
  end
  res += 0.001
  puts res.to_i
end
